app/src/main/java/com/example/cooky/base/BaseFragment.kt

```diff
@@ -31,8 +33,35 @@ abstract class BaseFragment<VB : ViewDataBinding, VM : ViewModel>() : Fragment()
             executePendingBindings()
         }
         setBindingVariables()
+        initView()
+        initData()
+        observeViewModel()
     }
 
-    open fun setBindingVariables() {
+    protected abstract fun initView()
+
+    protected open fun initData() {
+    }
+
+    protected abstract fun setBindingVariables()
+
+    protected open fun observeViewModel() = with(viewModel) {
+        isLoading.observe(viewLifecycleOwner, Observer { handleLoading(it) })
```

Comment:
Viết kiểu này cho đẹp.
```suggestion
        isLoading.observe(viewLifecycleOwner, Observer(::handleLoading)
```